src/Module/Api/Mastodon/FollowRequests.php

```diff
@@ -30,32 +28,39 @@ public static function init(array $parameters = [])
 	 * @param array $parameters
 	 * @throws HTTPException\BadRequestException
 	 * @throws HTTPException\ForbiddenException
+	 * @throws HTTPException\InternalServerErrorException
 	 * @throws HTTPException\NotFoundException
 	 * @throws HTTPException\UnauthorizedException
+	 * @throws \ImagickException
+	 *
 	 * @see https://docs.joinmastodon.org/methods/accounts/follow_requests#accept-follow
 	 * @see https://docs.joinmastodon.org/methods/accounts/follow_requests#reject-follow
 	 */
 	public static function post(array $parameters = [])
 	{
 		parent::post($parameters);
 
-		$Intro = DI::intro()->fetch(['id' => $parameters['id'], 'uid' => self::$current_user_id]);
+		$Introduction = DI::intro()->selectFirst(['id' => $parameters['id'], 'uid' => self::$current_user_id]);
 
-		$contactId = $Intro->{'contact-id'};
-
-		$relationship = new Mastodon\Relationship();
-		$relationship->id = $contactId;
+		$contactId = $Introduction->{'contact-id'};
 
 		switch ($parameters['action']) {
 			case 'authorize':
-				$Intro->confirm();
-				$relationship = Mastodon\Relationship::createFromContact(Contact::getById($contactId));
+				$Introduction->confirm();
+				DI::intro()->delete($Introduction);
+
+				$relationship = new Relationship($contactId, Contact::getById($contactId));
```

Comment:
```suggestion
				$relationship = DI::mstdnRelationship()->createFromContact($contactId);
```

Reply:
How did I miss that, I created the factory and all.